api/src/main/java/org/openmrs/module/fhir2/api/FhirDiagnosisService.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api;
+
+import ca.uhn.fhir.rest.api.server.IBundleProvider;
+import org.hl7.fhir.r4.model.Condition;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+
+public interface FhirDiagnosisService extends FhirService<Condition> {
+	
+	IBundleProvider searchDiagnoses(SearchParameterMap theParams);
```

Comment:
A better pattern for this that we've been slowly adopting is create a `SearchParams` builder that can be used to construct the `SearchParameterMap` and pass that between the `ResourceProvider` and the `*Service` class. See, for example, the [ConditionSearchParams](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/search/param/ConditionSearchParams.java). We should employ that pattern here.